portal/server/routes/middleware/validateMongoId/index.js

```diff
@@ -0,0 +1,20 @@
+const { isValidMongoId } = require('../../../validation/validate-ids');
+
+/**
+ * Redirects the user to '/not-found' if an invalid MongoDB '_id' path param is
+ * provided
+ * @param {object} req - The request object
+ * @param {object} res - The response object
+ * @param {Function} next - The next middleware function
+ */
+const validateMongoId = (req, res, next) => {
+  const { _id } = req.params;
+
+  if (!isValidMongoId(_id)) {
+    return res.redirect('/not-found');
+  }
```

Comment:
I don't know if this is in the scope of this ticket, but are we supposed to redirect to 'not-found'? Won't this render a page that says 'Page not found' but it's actually an error of the file not existing rather than the page not being found? I'd be expecting to see an error on the page saying "File does not exist" or something like that, but I'm not fully sure what the intended behaviour is

Reply:
I think the only case of this happening would be if the user manually entered the URL, so I think it's fair to redirect to `/not-found` in that case. There is precedent for this in `validateBank` which validates to `/not-found` when that check fails.

I think we do need to validate here to prevent the user entering something dodgy which then gets included in the upstream request to the portal-api.

Again, there is some inconsistency though where some validators at the UI server level will handle the result themselves, and others (such as `virusScanUpload`) will just set some `res.locals` value and let the controller handle the result. I guess that's more useful in the case where we want to show a specific page.

@ChristianMcc - might need your steer on this one. Would be good to discuss when you have time.

Reply:
Few points before I give a full answer:
- Definitely need to validate this as you said
- In the virusScanUpload etc. we mainly do this where we want to give custom error messages or errors that are likely to happen and need to stay on the same page

For this one I'm a little torn between routing to `/not-found` or not re-routing and just returning 'something went wrong' page. Given we should be confident in our system that we wouldn't give bad links then inclined to agree with Mike that this is only hit by manually entering a bad URL so I'll probably stick with this `not-found` approach!